core/dictgen/src/Scanner.cxx

```diff
@@ -101,6 +101,26 @@ RScanner::~RScanner ()
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Whether we can actually visit this declaration, i.e. if it is reachable
+/// via name lookup.
+///
+/// RScanner shouldn't touch decls for which this method returns false as we
+/// call Sema methods on those declarations. Those will fail in strange way as
+/// they assume those decls are already visible.
+///
+/// The main problem this is supposed to prevent is when we use C++ modules and
+/// have hidden declarations in our AST. Usually they can't be found as they are
+/// hidden from name lookup until their module is actually imported, but as the
+/// RecursiveASTVisitor is not supposed to be restricted by lookup limitations,
+/// it still reaches those hidden declarations.
+bool RScanner::shouldVisitDecl(clang::NamedDecl *D) {
+  if (auto M = D->getOwningModule()) {
```

Comment:
IIRC, `M` can be `nullptr` for some builtin decls (as per [D33366](https://reviews.llvm.org/D33366)).

Is this check here to check if we enabled runtime modules?

Reply:
I think you answered your question in the first sentence of you comment? If the owning module is a nullptr, we have a decl without a owned module and we can assume it's not from an unimported submodule.

Reply:
Yes, if we assume that all decls without an owning module set are visible.